iroh/src/endpoint.rs

```diff
@@ -634,68 +668,42 @@ impl Endpoint {
                 )
             })?;
 
-        debug!(
-            "connecting to {}: (via {} - {:?})",
-            node_id, addr, direct_addresses
-        );
+        debug!("connecting via {addr} - {direct_addresses:?}",);
+
+        let transport_config = options
+            .transport_config
+            .unwrap_or(self.static_config.transport_config.clone());
 
         // Start connecting via quinn. This will time out after 10 seconds if no reachable
         // address is available.
-        self.connect_quinn(node_id, alpn, addr, transport_config)
-            .await
-    }
 
-    #[instrument(
-        name = "connect",
-        skip_all,
-        fields(
-            me = %self.node_id().fmt_short(),
-            remote_node = node_id.fmt_short(),
-            alpn = %String::from_utf8_lossy(alpn),
-        )
-    )]
-    async fn connect_quinn(
-        &self,
-        node_id: NodeId,
-        alpn: &[u8],
-        addr: NodeIdMappedAddr,
-        transport_config: Arc<TransportConfig>,
-    ) -> Result<Connection> {
         debug!("Attempting connection...");
         let client_config = {
             let alpn_protocols = vec![alpn.to_vec()];
             let quic_client_config = tls::make_client_config(
                 &self.static_config.secret_key,
                 Some(node_id),
                 alpn_protocols,
+                Some(self.session_store.clone()),
                 self.static_config.keylog,
             )?;
             let mut client_config = quinn::ClientConfig::new(Arc::new(quic_client_config));
             client_config.transport_config(transport_config);
             client_config
         };
 
-        // TODO: We'd eventually want to replace "localhost" with something that makes more sense.
+        let server_name = &format!("{}.iroh.invalid", BASE32_DNSSEC.encode(node_id.as_bytes()));
```

Comment:
Could we add a comment that this is only used as a key in the TLS session store and has no other function?  (If that is indeed correct?  At least that is currently the only *reason* for this.)  Might help with future decisions that don't like this naming pattern anymore.

Reply:
Yep, there's no other reason for it yet, and I haven't seen changes to this break anything anywhere yet.